Library/brew.rb

```diff
@@ -22,32 +17,6 @@
   ARGV << ARGV.shift
 end
 
-if OS.mac?
-  # Check for bad xcode-select before other checks, because `doctor` and
-  # many other things will hang. Note that this bug was fixed in 10.9
-  if MacOS.version < :mavericks && MacOS.active_developer_dir == "/"
-    odie <<-EOS.undent
-      Your xcode-select path is currently set to '/'.
-      This causes the `xcrun` tool to hang, and can render Homebrew unusable.
-      If you are using Xcode, you should:
-        sudo xcode-select -switch /Applications/Xcode.app
-      Otherwise, you should:
-        sudo rm -rf /usr/share/xcode-select
-    EOS
-  end
-
-  # Check for user agreement of the Xcode license before permitting
-  # any other brew usage to continue. This prevents the situation where
-  # people are instructed to "please re-run as root via sudo" on brew commands.
-  # The check can only fail when Xcode is installed & the active developer dir.
-  if MacOS::Xcode.installed? && `/usr/bin/xcrun clang 2>&1` =~ /license/ && !$?.success?
-    odie <<-EOS.undent
-      You have not agreed to the Xcode license. Please resolve this by running:
-        sudo xcodebuild -license
-    EOS
-  end
-end
-
 case HOMEBREW_PREFIX.to_s
 when "/", "/usr"
```

Comment:
I think it's sufficient if we get just the bare minimum into shell code to get `update` working. We can then make incremental changes afterwards, which will be a lot easier to review and test.